.github/actions/composite/buildAndroidE2EAPK/action.yml

```diff
@@ -71,11 +71,17 @@ runs:
         echo "EXPENSIFY_PARTNER_USER_SECRET=${{ inputs.EXPENSIFY_PARTNER_USER_SECRET }}" >> ${{ inputs.PATH_ENV_FILE }}
         echo "EXPENSIFY_PARTNER_PASSWORD_EMAIL=${{ inputs.EXPENSIFY_PARTNER_PASSWORD_EMAIL }}" >> ${{ inputs.PATH_ENV_FILE }}
 
-    - name: Build APK
-      run: npm run ${{ inputs.PACKAGE_SCRIPT_NAME }}
-      shell: bash
+    - name: Build APK (retryable)
+      uses: nick-fields/retry@v3
       env:
         RUBYOPT: '-rostruct'
+      with:
+        retry_on: error
+        retry_wait_seconds: 60
+        timeout_minutes: 60
```

Comment:
I don't have much context, but this seems like a very long timeout. How long do the builds usually take?

Reply:
30-45 minutes

Reply:
Alright makes sense 👍